dataprofiler/profilers/column_profile_compilers.py

```diff
@@ -381,3 +381,28 @@ def profile(self):
             profile["statistics"] = \
                 self._profiles[TextProfiler.type].profile
         return profile
+
+    def diff(self, other, options=None):
+        """
+        Finds the difference between 2 compilers and returns the report
+
+        :param other: profile compiler finding the difference with this one.
+        :type other: UnstructuredCompiler
+        :param options: options to impact the results of the diff
+        :type options: dict
+        :return: difference of the profiles
+        :rtype: dict
+        """
+        # Call super for compiler instance check
+        diff_profile = super().diff(other, options)
+
+        # Iterate through profiles
+        all_profiles = set(self._profiles.keys()) & set(other._profiles.keys())
+        for key in all_profiles:
+            diff = self._profiles[key].diff(other._profiles[key], options)
+            if key == UnstructuredLabelerProfile.type:
+                diff_profile["data_label"] = diff
+            elif key == TextProfiler.type:
+                diff_profile["statistics"] = diff
```

Comment:
is this something that needs this abstraction? I'm not seeing the future need for it yet, but just pointing out. In structured I think it made more sense, but here we do a for loop and then specifically call out each.

thoughts?